config/network.go

```diff
@@ -0,0 +1,161 @@
+package config
+
+import (
+	"encoding/base64"
+	"fmt"
+	"strings"
+
+	"errors"
+
+	"github.com/pelletier/go-toml/v2"
+
+	"github.com/smartcontractkit/chainlink-testing-framework/utils/osutil"
+)
+
+const (
+	Base64NetworkConfigEnvVarName = "BASE64_NETWORK_CONFIG"
+)
+
+type NetworkConfig struct {
+	SelectedNetworks []string            `toml:"selected_networks"`
+	RpcHttpUrls      map[string][]string `toml:"RpcHttpUrls"`
+	RpcWsUrls        map[string][]string `toml:"RpcWsUrls"`
+	WalletKeys       map[string][]string `toml:"WalletKeys"`
+}
+
+func (n *NetworkConfig) ApplySecrets() error {
+	encodedEndpoints, err := osutil.GetEnv(Base64NetworkConfigEnvVarName)
+	if err != nil {
+		return err
+	}
+
+	err = n.ApplyBase64Enconded(encodedEndpoints)
+	if err != nil {
+		return fmt.Errorf("error reading network encoded endpoints: %w", err)
+	}
+
+	return nil
+}
+
+func (n *NetworkConfig) ApplyDecoded(configDecoded string) error {
+	if configDecoded == "" {
+		return nil
+	}
+
+	var cfg NetworkConfig
+	err := toml.Unmarshal([]byte(configDecoded), &cfg)
+	if err != nil {
+		return fmt.Errorf("error unmarshaling network config: %w", err)
+	}
+
+	err = n.ApplyOverrides(&cfg)
+	if err != nil {
+		return fmt.Errorf("error applying overrides from decoded network config file to config: %w", err)
+	}
+
+	return nil
+}
+
+func (n *NetworkConfig) ApplyBase64Enconded(configEncoded string) error {
+	if configEncoded == "" {
+		return nil
+	}
+
+	decoded, err := base64.StdEncoding.DecodeString(configEncoded)
+	if err != nil {
+		return err
+	}
+
+	return n.ApplyDecoded(string(decoded))
+}
+
+func (n *NetworkConfig) Validate() error {
+	if len(n.SelectedNetworks) == 0 {
+		return errors.New("selected_networks must be set")
+	}
+
+	upperCasedHttpUrls := make(map[string][]string)
+	for key := range n.RpcHttpUrls {
+		upperCasedHttpUrls[strings.ToUpper(key)] = n.RpcHttpUrls[key]
+	}
+	n.RpcHttpUrls = upperCasedHttpUrls
```

Comment:
So `Validate()` is mutating our config, can't it cause any issues with URLs?

Reply:
Ah, it's the key, then okay

Reply:
you are right, I will move it to dedicated method, my idea was that we shouldn't really care whether the users input network as `Polygon_Mumbai` or `polygon_mumbai` or `POlygoN_MumbAI`, but I think it's better to move that code to sth like `UpperCaseNetworkNames()`